File: src/types/voice.ts
/**
 * Voice 数据类型 — 对应架构文档 §2.1 表5
 */

export interface VoiceRecord {
  id: number
  audio_path: string
  duration_seconds: number
  transcript: string | null
  ai_summary: string | null
  ai_todos: string | null   // JSON 数组字符串
  status: 'recording' | 'processing' | 'done' | 'error'
  created_at: string
}

File: src-tauri/src/models/voice.rs
use serde::Serialize;

/// 语音记录（对应 voice_records 表）
#[derive(Debug, Clone, Serialize)]
pub struct VoiceRecord {
    pub id: i64,
    pub audio_path: String,
    pub duration_seconds: i32,
    pub transcript: Option<String>,
    pub ai_summary: Option<String>,
    pub ai_todos: Option<String>,
    pub status: String,
    pub created_at: String,
}
